### PR TITLE
feat: adds new type and output metadata on pieces

### PR DESCRIPTION
**Pull request by RasmusAlbrektsen**
This PR introduces new metadata that is placed on pieces. This can be used to more specifically differ between pieces without combining many different properties to do so. For now this is mostly implemented so we can use it in the frontend to render pieces.

### CODE REVIEW

**Review by andr9528 — APPROVED** [review]

**Review by LindvedKrvang — APPROVED** [review]
FYI: The `type` will also be used by Actions

File: src/tv2-common/helpers/graphics/pilot/PilotGraphicGenerator.ts [hunk 1]
```diff
@@ -184,6 +191,33 @@ export abstract class PilotGraphicGenerator extends Graphic {
...
+				return Tv2PieceType.GRAPHICS
+			default:
+				return Tv2PieceType.GRAPHICS
+		}
+	}
+
+	protected getTv2OutputLayer(): Tv2OutputLayer | undefined {
+		switch (this.engine) {
+			case 'WALL':
+				return undefined
```
Comment (KvelaGorrrrnio) [inline src/tv2-common/helpers/graphics/pilot/PilotGraphicGenerator.ts:210]: The WALL case can be removed or moved down to the default case without changing the behaviour.

**Review by KvelaGorrrrnio — APPROVED** [review]

File: src/tv2-constants/tv2-piece-type.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export enum Tv2PieceType {
+	CAMERA = 'CAMERA',
+	REPLAY = 'REPLAY',
+	REMOTE = 'REMOTE',
+	GRAPHICS = 'GRAPHICS',
+	OVERLAY_GRAPHICS = 'OVERLAY_GRAPHICS',
+	SPLIT_SCREEN = 'SPLIT_SCREEN',
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/tv2-constants/tv2-piece-type.ts:7]: After a talk between the developers this was decided to be kept as `SPLIT_SCREEN`.

File: src/tv2-constants/tv2-piece-type.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export enum Tv2PieceType {
+	CAMERA = 'CAMERA',
+	REPLAY = 'REPLAY',
+	REMOTE = 'REMOTE',
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/tv2-constants/tv2-piece-type.ts:4]: Same as with `SPLIT_SCREEN`, a developer conversation ended with choosing `REMOTE`.

File: src/tv2-common/helpers/graphics/pilot/PilotGraphicGenerator.ts (hunk 1 above)
Comment (RasmusAlbrektsen) [reply src/tv2-common/helpers/graphics/pilot/PilotGraphicGenerator.ts]: I have removed the case like you suggested.

File: src/tv2-constants/tv2-output-layer.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export enum Tv2OutputLayer {
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/tv2-constants/tv2-output-layer.ts:1]: I did think about this, but kept it as so to also signify that the metadata is different from the similar-sounding Piece properties that exist as well.

**Pull request merged**